app/controllers/reservations_controller.rb

```diff
@@ -369,20 +369,8 @@ def current
     render 'current_reservations'
   end
 
-  # two paths to create receipt emails for checking in and checking out items.
-  def checkout_email
-    if UserMailer.checkout_receipt(@reservation).deliver
-      redirect_to :back
-      flash[:notice] = 'Successfully delivered receipt email.'
-    else
-      redirect_to @reservation
-      flash[:error] = 'Unable to deliver receipt email. Please contact '\
-        'administrator for more support.'
-    end
-  end
-
-  def checkin_email
-    if UserMailer.checkin_receipt(@reservation).deliver
+  def send_receipt
+    if UserMailer.reservation_status_update(@reservation).deliver
       redirect_to :back
```

Comment:
This is no good since if we call this from the `receipt` method that follows the `checkout` or `checkin` methods, then it tries to run `checkout` again resulting in a `RecordNotFound` exception. Why don't we redirect to the reservation in both cases?